pkg/block/index.go

```diff
@@ -673,14 +677,14 @@ func rewrite(
 			meta.Stats.NumSamples += uint64(chk.Chunk.NumSamples())
 		}
 
-		for _, l := range s.lset {
+		s.lset.Range(func(l labels.Label) {
 			valset, ok := values[l.Name]
 			if !ok {
 				valset = stringset{}
 				values[l.Name] = valset
```

Comment:
I think we still access `l.Name` here

Reply:
Sorry i dont understand! Can you rephrase please?